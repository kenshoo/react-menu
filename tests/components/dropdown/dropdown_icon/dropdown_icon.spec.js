import React from "react";
import renderer from "react-test-renderer";
import DropdownIcon from "../../../../src/components/dropdown/dropdown_icon/dropdown_icon";

describe("DropdownIcon component", () => {
  test("default snapshot", () => {
    const cmp = renderer.create(<DropdownIcon />);
    expect(cmp).toMatchSnapshot();
  });

  test("className prop", () => {
    const cmp = renderer.create(<DropdownIcon clasName="TEST" />);
    expect(cmp).toMatchSnapshot();
  });

  test("arrow direction - left", () => {
    const cmp = renderer.create(<DropdownIcon direction="left" />);
    expect(cmp).toMatchSnapshot();
  });

  test("arrow direction - bottom", () => {
    const cmp = renderer.create(<DropdownIcon direction="bottom" />);
    expect(cmp).toMatchSnapshot();
  });

  test("arrow direction - right", () => {
    const cmp = renderer.create(<DropdownIcon direction="right" />);
    expect(cmp).toMatchSnapshot();
  });

  test("arrow direction - unknown", () => {
    const cmp = renderer.create(<DropdownIcon direction="TEST" />);
    expect(cmp).toMatchSnapshot();
  });

  test("disabled prop", () => {
    const cmp = renderer.create(<DropdownIcon isDisabled />);
    expect(cmp).toMatchSnapshot();
  });

  test("disabledClassName prop", () => {
    const cmp = renderer.create(
      <DropdownIcon isDisabled disabledClassName="TEST" />
    );
    expect(cmp).toMatchSnapshot();
  });
});
