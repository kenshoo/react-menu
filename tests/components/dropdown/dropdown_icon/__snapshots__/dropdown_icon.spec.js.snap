// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownIcon component arrow direction - bottom 1`] = `
<svg
  attr={
    Object {
      "viewBox": "0 0 448 512",
    }
  }
  className="undefined icon"
  fill="currentColor"
  height="1em"
  stroke="currentColor"
  strokeWidth="0"
  style={
    Object {
      "color": undefined,
    }
  }
  viewBox="0 0 448 512"
  width="1em"
>
  <path
    d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
  />
</svg>
`;

exports[`DropdownIcon component arrow direction - left 1`] = `
<svg
  attr={
    Object {
      "viewBox": "0 0 320 512",
    }
  }
  className="undefined icon"
  fill="currentColor"
  height="1em"
  stroke="currentColor"
  strokeWidth="0"
  style={
    Object {
      "color": undefined,
    }
  }
  viewBox="0 0 320 512"
  width="1em"
>
  <path
    d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
  />
</svg>
`;

exports[`DropdownIcon component arrow direction - right 1`] = `
<svg
  attr={
    Object {
      "viewBox": "0 0 320 512",
    }
  }
  className="undefined icon"
  fill="currentColor"
  height="1em"
  stroke="currentColor"
  strokeWidth="0"
  style={
    Object {
      "color": undefined,
    }
  }
  viewBox="0 0 320 512"
  width="1em"
>
  <path
    d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
  />
</svg>
`;

exports[`DropdownIcon component arrow direction - unknown 1`] = `
<svg
  attr={
    Object {
      "viewBox": "0 0 448 512",
    }
  }
  className="undefined icon"
  fill="currentColor"
  height="1em"
  stroke="currentColor"
  strokeWidth="0"
  style={
    Object {
      "color": undefined,
    }
  }
  viewBox="0 0 448 512"
  width="1em"
>
  <path
    d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
  />
</svg>
`;

exports[`DropdownIcon component className prop 1`] = `
<svg
  attr={
    Object {
      "viewBox": "0 0 448 512",
    }
  }
  className="undefined TEST icon"
  fill="currentColor"
  height="1em"
  stroke="currentColor"
  strokeWidth="0"
  style={
    Object {
      "color": undefined,
    }
  }
  viewBox="0 0 448 512"
  width="1em"
>
  <path
    d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
  />
</svg>
`;

exports[`DropdownIcon component default snapshot 1`] = `
<svg
  attr={
    Object {
      "viewBox": "0 0 448 512",
    }
  }
  className="undefined icon"
  fill="currentColor"
  height="1em"
  stroke="currentColor"
  strokeWidth="0"
  style={
    Object {
      "color": undefined,
    }
  }
  viewBox="0 0 448 512"
  width="1em"
>
  <path
    d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
  />
</svg>
`;

exports[`DropdownIcon component disabled prop 1`] = `
<svg
  attr={
    Object {
      "viewBox": "0 0 448 512",
    }
  }
  className="undefined icon disabled"
  fill="currentColor"
  height="1em"
  stroke="currentColor"
  strokeWidth="0"
  style={
    Object {
      "color": undefined,
    }
  }
  viewBox="0 0 448 512"
  width="1em"
>
  <path
    d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
  />
</svg>
`;

exports[`DropdownIcon component disabledClassName prop 1`] = `
<svg
  attr={
    Object {
      "viewBox": "0 0 448 512",
    }
  }
  className="undefined icon TEST"
  fill="currentColor"
  height="1em"
  stroke="currentColor"
  strokeWidth="0"
  style={
    Object {
      "color": undefined,
    }
  }
  viewBox="0 0 448 512"
  width="1em"
>
  <path
    d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
  />
</svg>
`;
